alphadia/planning.py

```diff
@@ -340,36 +342,19 @@ def run(
                 psm_df.to_parquet(psm_location, index=False)
                 frag_df.to_parquet(frag_location, index=False)
 
-                workflow.reporter.log_string(f"Finished workflow for {raw_name}")
-                workflow.reporter.context.__exit__(None, None, None)
-                del workflow
-
-            except peptidecentric.CalibrationError:
-                # get full traceback
-                logger.error(
-                    f"Search for {raw_name} failed as not enough precursors were found for calibration"
-                )
-                logger.error("This can have the following reasons:")
-                logger.error(
-                    "   1. The sample was empty and therefore nor precursors were found"
-                )
-                logger.error("   2. The sample contains only very few precursors.")
-                logger.error(
-                    "      For small libraries, try to set recalibration_target to a lower value"
-                )
-                logger.error(
-                    "      For large libraries, try to reduce the library size and reduce the calibration MS1 and MS2 tolerance"
-                )
-                logger.error(
-                    "   3. There was a fundamental issue with search parameters"
-                )
+            except CustomError as e:
+                _log_exception_event(e, raw_name, workflow)
                 continue
```

Comment:
just confirming: in the case of a business error, we continue, on other errors we fail, correct?

but if for 100 files, 2 have business errors, then the remaning 98 will be quanted and success will be reported, right? how will the user know that 2 files had errors?

Reply:
Yes, exactly. The two files will be appear as all zero rows from the stat output.